js/reducers/passageReducer.js

```diff
@@ -100,4 +101,14 @@ function passageReducer(state = initialState, action) {
   }
 }
 
-export default passageReducer;
+function stateSavingPassageReducer(state, action) {
+  if (!state) {
+    state = assignToEmpty(defaultState, restoreState());
```

Comment:
Well, I preserved the signature when I wrapped the function, and moreover all the tests passed... One of the nice things about reducers being simple functions is that it's straightforward to compose them together like this.

The problem with restoring the state when we define initial state is that it happens once, at load time. That would work in the browser, but not really for testing since the tests don't reload the file in between runs. And if they did that would the mean we wouldn't have an easy way to persist the cookie value across loads.